fury/ui/elements.py

```diff
@@ -3052,3 +3054,396 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
+
+    def _get_size(self):
+        return self.canvas.size
+
+    def _set_position(self, coords):
+        """Set the lower-left corner position of this UI component.
+
+        Parameters
+        ----------
+        coords: (float, float)
+            Absolute pixel coordinates (x, y).
+        """
+        self.canvas.position = coords
+
+    def resize(self, size):
+        """Resize the UI.
+        """
+        pass
+
+    @property
+    def current_mode(self):
+        return self._current_mode
+
+    @current_mode.setter
+    def current_mode(self, mode):
+        self.update_button_icons(mode)
+        self._current_mode = mode
+        if mode is not None:
+            self.mode_text.message = f"Mode: {mode}"
+
+    def cal_min_boundary_distance(self, position):
+        """Calculate the minimum distance between the current position and canvas boundary.
+
+        Parameters
+        ----------
+        position: (float,float)
+            current position of the shape.
+
+        Returns
+        -------
+        float
+            Minimum distance from the boundary.
+        """
+        distance_list = []
+        # calculate distance from element to left and lower boundary
+        distance_list.extend(position - self.canvas.position)
+        # calculate distance from element to upper and right boundary
+        distance_list.extend(self.canvas.position + self.canvas.size - position)
+
+        return min(distance_list)
+
+    def draw_shape(self, shape_type, current_position, in_process=False):
+        """Draws the required shape at the given position.
+
+        Parameters
+        ----------
+        shape_type: string
+            Type of shape - line, quad, circle.
+        current_position: (float,float)
+            Lower left corner position for the shape.
+        in_process: bool, optional
+            Checks whether in process or not.
+        """
+        if not in_process:
+            shape = DrawShape(shape_type=shape_type, drawpanel=self,
+                              position=current_position)
+            if shape_type == "circle":
+                shape.max_size = self.cal_min_boundary_distance(current_position)
+            self.shape_list.append(shape)
+            self.current_scene.add(shape)
+            self.canvas.add_element(shape, current_position - self.canvas.position)
+
+        else:
+            current_shape = self.shape_list[-1]
+            size = current_position - current_shape.position
+            current_shape.resize(size)
+
+    def update_button_icons(self, current_mode):
+        """Updates the button icon.
+
+        Parameters
+        ----------
+        last_mode: string
+            Previous mode of the UI.
+        current_mode: string
+            Current mode of the UI.
+        """
+        for btn in self.mode_panel._elements[1:]:
+            if btn.icon_names[0] == current_mode:
+                btn.next_icon()
+            elif btn.current_icon_id == 1:
```

Comment:
==1 ? what does it mean?

Reply:
These buttons basically have two icons 
0th index/id - static icon
1st index/id - pressed icon
So I am updating the icon of the `current_mode` button 
and checking if any other button has its icon set to the pressed mode (which is 1st index/id) then updating it too.

Reply:
make sense. thanks!